src/common/regions.ts

```diff
@@ -119,16 +141,62 @@ const counties: County[] = chain(state_county_map_dataset)
   })
   .value();
 
+const countiesByFips = fromPairs(
+  counties.map(county => [county.fipsCode, county]),
+);
+
+// Seattle-Tacoma, WA -> seattle-tacoma_wa
+function urlFormat(name: string) {
+  return name
+    .toLowerCase()
+    .split(', ')
+    .map(part => deburr(words(part).join('-')))
+    .join('_');
+}
+
+const metroAreas: MetroArea[] = msaDefinitions.msa_regions.map(metroArea => {
+  // TODO: Handle counties with FIPS: 36005, 360047, 36081, 36085, those
+  // are part of MSAs, but not in the countiesByFips map
+  const metroAreaCounties: County[] = metroArea.countyFips
+    .map(countyFips => countiesByFips[countyFips])
+    .filter(county => county);
+
+  const totalPopulation = sum(
+    metroAreaCounties.map(county => county.population),
+  );
+
+  return new MetroArea(
+    metroArea.csaTitle,
+    urlFormat(metroArea.csaTitle),
+    metroArea.cbsaCode,
+    totalPopulation,
+    metroAreaCounties,
+  );
```

Comment:
Could we have this code in the file that produces the json? Feels a bit unnecessary that we have to regenerate this every time, when its essentially a static calculation